logd/src/logd/proto_converter.cpp

```diff
@@ -59,7 +60,20 @@ ProtoConverter::log_message_to_proto(const LogMessage& message, ProtoLogMessage&
     proto.set_service(message.service());
     proto.set_component(message.component());
     proto.set_level(convert_level(message.level()));
-    proto.set_payload(message.payload());
+    const std::string &payload = message.payload();
+    vespalib::Utf8Reader reader(payload.c_str(), payload.size());
```

Comment:
Consider moving the filtering to a helper function to make the code easier to read.

Reply:
please take another look now